unit_tests/test_spectrographs.py

```diff
@@ -1,11 +1,67 @@
 """
 Module to test spectrograph read functions
 """
+from pathlib import Path
 import os
 
 from IPython import embed
 
 from pypeit import spectrographs
+from pypeit.spectrographs.util import load_spectrograph
+
+
+# test that we can find the correct files
+def generic_file_search_test(spec, root, nfiles):
+    devroot = os.getenv('PYPEIT_DEV')
+    assert devroot is not None, 'PYPEIT_DEV not defined'
+    _root = Path(devroot).absolute() / 'RAW_DATA' / spec / root
+    assert _root.is_dir(), f'{_root} is not a directory'
+    files = load_spectrograph(spec).find_raw_files(_root)
+    assert len(files) == nfiles, f'Found {len(files)} {spec} files in {_root}, not {nfiles}'
+
+def test_filesearch():
+    #         Spectrograph name, setup directory, expected number of files
+    args = [['aat_uhrf', '3875', 5],
```

Comment:
This is great, but when someone adds a new setup, should we request that this test is updated as well, or do you think this is sufficient for testing purposes?

Reply:
Good question.  As it is, this list is incomplete.

I only picked one setup per spectrograph; so I don't think we need to add a new entry for each new setup.  But it would be useful to add a new entry for each new spectrograph.  Having said that, there are other similar tests where we are incomplete (e.g., `test_load_images.py`)

The other way this is incomplete is that many of the current spectrographs (e.g., `vlt_xshooter`) are not included.  This is because the directory name in the dev-suite doesn't match the spectrograph name.  Instead of adding this complication to the test, I think it should be on our to-do list to restructure the dev-suite so that *all* instruments follow the `spectrograph/setup` nomenclature, where `spectrograph` is identical to the pypeit-specific spectrograph name.

Reply:
I think that's totally fine. There should be enough diversity in the current instrument suite to make sure this remains well-tested, anyway. Something good to keep in mind though 👍